### PR TITLE
Add support for generating tsconfig

### LINKED ISSUE #8: Add support for generating tsconfig

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
- fixes #8

### CODE REVIEW

File: plop-templates/tsconfig.json.hbs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Generated by seam-plop
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "allowJs": true,
+    "skipLibCheck": true,
+    "strict": false,
```
[the remark this answers is not in the document]
Comment (seveibar) [reply plop-templates/tsconfig.json.hbs:8]: looked into this a bunch today in the context of converting some projects over. I think strict mode should be a requirement for libraries and recommended for APIs, but not applications (especially frontend applications). We should always default to `"strict": true` A bit on my process:

I converted Seam for Homes to strict mode and looked at all the errors, I think the typing would fix a lot of errors, but it was also clear that a lot of functions would have pretty large signatures, and some of the errors that came up only in strict mode were difficult to debug because of complex type involvement. There were also a lot of errors (195), I found some examples where I thought the addition of types was good/bad (some examples below but not showing any particular point)

image

image

I want to be careful to make recommendations that don't impact velocity, I think strict mode would have helped with a lot of bugs on seam for homes, and ultimately it's not preventing the use of `any` to quickly get around issues and write bad code, but it makes it more clear where the programmer is taking a shortcut.

TLDR; strict mode is probably good
